### ISSUE TITLE
[BUG] Incorrect rendering of spoiler/link markdown

### ISSUE THREAD

**Issue by bohrabhijeet** [laurencedawson/sync-for-reddit#1490]
**Reported here**:
https://www.reddit.com/r/redditsync/comments/vh7knw

I've rephrased the title as it's not exactly what the bug is

> **Description**
> 
> In prior versions, I remember that spoiler links used to work, but now they do not. I think they are of [this form](spoiled haha).
> 
> **Scenario** (steps to reproduce)
> 
> 1. Go to [this comment](https://www.reddit.com/r/SpaceXLounge/comments/vgpkfh/we_got_a_leaked_look_at_nasas_future_moon/id3xn9h/);
> 2. Tap on the spoiler link. It should open the spoiler text inside, but it doesn't!
> 
> **Result(s)**
> 
> Nothing happens when tapping!
> 
> **Device information**
> 
>     Sync version: v22.06.17-16:57    
>     Sync flavor: pro    
>     
>     View type: Slides    
>     Push enabled: true    
>     
>     Device: r9q    
>     Model: samsung SM-G990B    
>     Android: 12
> 
> EDIT: It should be noted that Sync recognizes the spoiler text somehow, since the post preview on the subreddit says "Spoiler hidden". I don't know if it analyzes the submission content or it's an heuristic based on the title, though.

TL;DR: Sync is rendering `[lorem](ipsum)` as a link where _ipsum_ is not a hyperlink, but text.

On the official Reddit client:
<img src="https://user-images.githubusercontent.com/34155181/174777046-74ad72a2-5a70-4692-801e-c5d30255d1ec.png" width="480" height="1040" />
